### PR TITLE
[CI] switch vst to flocq

### PR DESCRIPTION
**Pull request by Alizter**
We switch the VST to flocq4 and delete the old flocq3 job since it was only around for VST.

### CODE REVIEW

File: Makefile.ci [hunk 1]
```diff
@@ -37,7 +37,6 @@ CI_TARGETS= \
     ci-fiat_crypto_legacy \
     ci-fiat_crypto_ocaml \
     ci-fiat_parsers \
-    ci-flocq3 \
```
Comment (SkySkimmer) [inline Makefile.ci:40]: also delete the script and basic-overlay section

Comment (Alizter) [reply Makefile.ci:40]: Done

File: dev/ci/ci-basic-overlay.sh [hunk 2]
```diff
@@ -104,11 +104,6 @@ project coqhammer "https://github.com/lukaszcz/coqhammer" "master"
 ########################################################################
 project geocoq "https://github.com/GeoCoq/GeoCoq" "master"
 
-########################################################################
-# Flocq 3
-########################################################################
-project flocq3 "https://gitlab.inria.fr/flocq/flocq" "flocq-3"
-
 ########################################################################
 # Flocq (master)
```
Comment (Zimmi48) [inline dev/ci/ci-basic-overlay.sh:108]: Maybe the parenthesis is not necessary anymore.

**Review by Zimmi48 — APPROVED** [review]

File: dev/ci/ci-basic-overlay.sh (hunk 2 above)
Comment (Alizter) [reply dev/ci/ci-basic-overlay.sh]: Done.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**